net.solarnetwork.node.weather.ibm.wc/src/net/solarnetwork/node/weather/ibm/wc/BasicWCClient.java

```diff
@@ -0,0 +1,246 @@
+
+	/** The default value for the {@code units} property. */
+	public static final String DEFAULT_UNITS = "e";
+
+	public static final boolean DEFAULT_COMPRESSION = true;
+
+	private String baseUrl;
+	private String dailyForecastUrl;
+	private String hourlyForecastUrl;
+	private String format;
+	private String language;
+	private String units;
+
+	public BasicWCClient() {
+		this.baseUrl = DEFAULT_BASE_URL;
+		this.dailyForecastUrl = DEFAULT_DAILY_FORECAST_URL;
+		this.hourlyForecastUrl = DEFAULT_HOURLY_FORECAST_URL;
+		this.format = DEFAULT_FORMAT;
+		this.language = DEFAULT_LANGUAGE;
+		this.units = DEFAULT_UNITS;
+
+		// set default compression
+		this.setCompress(DEFAULT_COMPRESSION);
+	}
+
+	private String getURL(String requestURL, String location, String apiKey, String datumPeriod) {
+
+		return this.baseUrl + "/"
+				+ String.format(requestURL, datumPeriod, location, language, format, units, apiKey);
+	}
+
+	/**
+	 * For the daily forecast, each part of the data (ie. max temp, min temp,
+	 * narrative) is an array, with each entry corresponding with a day of the
+	 * week. The more detailed features in "daypart" such as cloud cover are
+	 * split into day and night, meaning two entries for each day.
+	 */
+	private BigDecimal parseBigDecimal(JsonNode n) {
+		if ( n.isNull() ) {
+			return null;
+		}
+		return new BigDecimal(n.asLong());
+	}
+
+	private LocalTime parseTimeValue(JsonNode n) {
+		if ( n == null || n.isNull() ) {
+			return null;
+		}
+		return new LocalTime(n.asLong() * 1000);
+
+	}
+
+	private Integer parseInt(JsonNode n) {
+		if ( n.isNull() ) {
+			return null;
+		}
+		return n.asInt();
+	}
+
+	//TODO valid time UTC
+	@Override
+	public Collection<GeneralDayDatum> readDailyForecast(String locationIdentifier, String apiKey,
+			DailyDatumPeriod datumPeriod) {
+
+		if ( locationIdentifier == null ) {
+			return null;
+		}
+		final List<GeneralDayDatum> result = new ArrayList<GeneralDayDatum>();
+		final String url = getURL(this.dailyForecastUrl, locationIdentifier, apiKey,
+				datumPeriod.toString());
+		JsonNode root;
+		try {
+			log.debug("opening IBM Weather API connection at URL {}", url);
+			root = getObjectMapper().readTree(jsonGET(url));
+		} catch ( IOException e ) {
+			log.warn("Error reading WC URL [{}]: {}", url, e.getMessage());
+			return result;
+		}
+
+		for ( int i = 0; i < root.get("sunriseTimeUtc").size(); i++ ) {
+
+			GeneralDayDatum current = new GeneralDayDatum();
+			JsonNode data = root.get("sunriseTimeUtc");
+			if ( data.isArray() ) {
+
+				current.setSunrise(parseTimeValue(data.get(i)));
+			}
+			data = root.get("sunsetTimeUtc");
+			if ( data.isArray() ) {
+				current.setSunset(parseTimeValue(data.get(i)));
+
+			}
+			data = root.get("moonriseTimeUtc");
+			if ( data.isArray() ) {
+				current.setMoonrise(parseTimeValue(data.get(i)));
+			}
+			data = root.get("moonsetTimeUtc");
+			if ( data.isArray() ) {
+				current.setMoonset(parseTimeValue(data.get(i)));
```

Comment:
All sun/moon rise/set values must use the appropriate time zone of the data... see previous comment on parsing local times.